File: app/src/main/java/com/example/win/easy/dagger/module/UtilModule.java
package com.example.win.easy.dagger.module;

import com.example.win.easy.value_object.VOUtil;
import com.example.win.easy.web.DTOUtil;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class UtilModule {

    @Provides
    @Singleton
    static DTOUtil provideDTOUtil(){
        return new DTOUtil();
    }

    @Provides
    @Singleton
    static VOUtil provideVOUtil(){
        return new VOUtil();
    }
}

File: app/src/main/java/com/example/win/easy/dagger/module/UtilModule.kt
package com.example.win.easy.dagger.module

import com.example.win.easy.parser.interfaces.FilenameParser
import com.example.win.easy.recognization.Image
import com.example.win.easy.view.ImageService
import com.example.win.easy.web.DTOUtil
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
object UtilModule {
    @JvmStatic
    @Provides
    @Singleton
    fun provideDTOUtil(filenameParser: FilenameParser<Char?>?): DTOUtil {
        return DTOUtil(filenameParser)
    }
}